sdk/inc/azure/core/internal/az_http_internal.h

```diff
@@ -70,7 +70,7 @@ typedef struct
  */
 AZ_NODISCARD AZ_INLINE _az_http_policy_telemetry_options _az_http_policy_telemetry_options_default()
 {
-  return (_az_http_policy_telemetry_options){ .os = AZ_SPAN_FROM_STR("Unknown OS") };
+  return (_az_http_policy_telemetry_options){ 0 };
```

Comment:
Should the default option value have some "placeholder" component_name? Like "package-name"?

Reply:
No, then assert will never work.
Previous commits here had no assert, and would go with just `azsdk-c/1.3.0-beta.1` when the component name is empty. But assert was suggested, and I think it is better.